src/context/service-worker-context.tsx

```diff
@@ -9,8 +25,32 @@ export const ServiceWorkerContext = createContext({
 export const ServiceWorkerProvider = ({ children }): JSX.Element => {
   const [isServiceWorkerRegistered, setIsServiceWorkerRegistered] = useState(false)
 
+  const windowLocation = translateIpfsRedirectUrl(window.location.href)
+
   useEffect(() => {
     if (isServiceWorkerRegistered) {
+      /**
+       * The service worker is registered, now we need to check for "helia-sw" and origin isolation support
+       */
+      if (windowLocation.href !== window.location.href) {
+      /**
+       * We're at a domain with ?helia-sw=, we can reload the page so the service worker will
+       * capture the request
+       */
+        window.location.replace(windowLocation.href)
```

Comment:
this check could be more explicit for future devs, but `translateIpfsRedirectUrl` should make understanding this easy.